code/modules/surgery/organs/organ.dm

```diff
@@ -248,13 +248,32 @@ var/list/organ_cache = list()
 		return
 	switch (severity)
 		if (1.0)
-			take_damage(0,20)
+			take_damage(20,0)
 			return
 		if (2.0)
-			take_damage(0,7)
+			take_damage(7,0)
 			return
 		if(3.0)
-			take_damage(0,3)
+			take_damage(3,0)
+
+/obj/item/organ/internal/emp_act(severity)
+	if(robotic == ORGAN_ROBOT)
+		switch (severity)
+			if (1.0)
+				take_damage(20,1)
+			if (2.0)
+				take_damage(7,1)
+			if(3.0)
+				take_damage(3,1)
+	else if(robotic == ORGAN_ASSISTED)
+		take_damage(11,1)
+
+/obj/item/organ/internal/heart/emp_act(intensity)
+	if(owner && robotic == 2)
+		owner.heart_attack = 1
+		owner.visible_message("<span class='danger'>[owner] clutches their chest and gasps!</span>","<span class='userdanger'>You clutch your chest in pain!</span>")
+	else if(owner && robotic == 1)
```

Comment:
@Crazylemon64 

Is what you said still relevant. As in, does it still need to be addressed?